pkg/controller/scyllacluster/resource.go

```diff
@@ -648,17 +651,47 @@ func stringOrDefault(str, def string) string {
 	return def
 }
 
-func MakeServiceAccount(sc *scyllav1.ScyllaCluster) *corev1.ServiceAccount {
+func MakeServiceAccount(ctx context.Context, sc *scyllav1.ScyllaCluster, serviceAccounts map[string]*corev1.ServiceAccount, coreClient corev1client.CoreV1Interface) (*corev1.ServiceAccount, error) {
+	var annotations map[string]string
+	name := naming.MemberServiceAccountNameForScyllaCluster(sc.Name)
+
+	// In 1.7.0 we started managing ServiceAccounts. Because previously these were created by users
+	// there's a high chance that they don't match our selector, hence are not available in serviceAccounts.
+	//
+	// We need to copy annotations stored in user SA because these may control integrations with k8s provider services
+	// like IAM management.
+	// Live call happens only once - when SA is being adopted. It can be removed in 1.8.0 because all SAs will be already
```

Comment:
Why do you need a live call?
I was thinking to:
- before calling this function list the cache and extend the map with the missing SA
- set RV, so ApplyServiceAccount work correctly with the input

Reply:
replaced live call with cache fetch. No need to ammend the fetched resource since we only need to copy annotations.